core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -191,7 +143,7 @@ private boolean isSameColumnSchemaList(List<ColumnSchema> indexFileColumnList,
       return detailedBlocklets;
     }
     Set<TableBlockIndexUniqueIdentifier> identifiers =
-        getTableBlockIndexUniqueIdentifiers(segment);
+        getTableBlockIndexUniqueIdentifiers(segment, this.getCarbonTable().getTableUniqueName());
```

Comment:
I think it is better to move getTableBlockIndexUniqueIdentifiers function to `CarbonTable`

Reply:
Can do that. But..

getTableBlockIndexUniqueIdentifiers is depednent on segmentMap of BlockletDataMapFactory. DeCoupling segmentMap from BlockletDataMapFactory needs more effrot. can handle in separate PR.